runtime/v2/runc/v2/service.go

```diff
@@ -197,7 +201,6 @@ func (s *service) StartShim(ctx context.Context, id, containerdBinary, container
 		}
 		return "", err
 	}
-	defer socket.Close()
```

Comment:
```
/run/containerd/s/35c0ec7f5eef15eb6fefae5675ffece5334b2f93eab1eacc9519006937b03db0: timeout: unknown
```

The case is from runc.v1 shimv2 binary. The socket.Close() should apply for runtime/v2/runc/v1/service.go